stdlib/public/runtime/ImageInspectionStatic.cpp

```diff
@@ -0,0 +1,54 @@
+//===-- ImageInspectionStatic.cpp -------------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// Implementation of functions to read data sections from static executable.
+//
+//===----------------------------------------------------------------------===//
+
+#include "ImageInspection.h"
+
+// Currently only tested on linux but should work for any ELF platform
+#if defined(__ELF__) && defined(__linux__)
+
+// These are defined in swift_sections.S
+// Used in swift_sections.S to mark the start of a section
+struct SectionInfo {
+  uint64_t size;
+  const uint8_t data[0];
```

Comment:
Is the following ok? I wanted to keep SectionInfo a struct as it will also be used in the PR which fixes the issue with libs that are dlopen()'d after startup. 
```

extern const uint64_t protocolConformancesStart asm(".swift2_protocol_conformances_start");
extern const uint64_t typeMetadataStart asm(".swift2_type_metadata_start");

struct SectionInfo {
  uint64_t size;
  const char *data;
};

static SectionInfo
_getSection(const uint64_t *section) {
  SectionInfo info;

  info.size = *section;
  info.data = reinterpret_cast<const char *>(section + 1);
  return info;
}

void
swift::initializeProtocolConformanceLookup() {
  auto protocolConformances = _getSection(&protocolConformancesStart);
  addImageProtocolConformanceBlockCallback(protocolConformances.data,
                                           protocolConformances.size);
}
```